content/en/blog/_posts/2023-11-1-Gateway-API-Future.md

```diff
@@ -0,0 +1,90 @@
+
+The ability to configure these timeouts adds a new layer of reliability to your Kubernetes services.
+Whether it's ensuring client requests are processed within a specified timeframe or managing backend service communications, Gateway API's Timeouts offer the control and predictability you need.
+
+To get started, you can define timeouts in your HTTPRoute Rules using the Timeouts field, specifying their type as Duration.
+A zero-valued timeout (`0s`) disables the timeout, while a valid non-zero-valued timeout should be at least 1ms.
+
+Here's an example of setting request and backendRequest timeouts in an HTTPRoute:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: timeout-example
+spec:
+  parentRefs:
+  - name: example-gateway
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /timeout
+    timeouts:
+      request: 10s
+      backendRequest: 2s
+    backendRefs:
+    - name: timeout-svc
+      port: 8080
+```
+
+In this example, a `request` timeout of 10 seconds is defined, ensuring that client requests are processed within that timeframe.
+Additionally, a 2-second `backendRequest` timeout is set for individual requests from the Gateway to a backend service called timeout-svc.
+
+These new HTTPRoute Timeouts provide Kubernetes users with more control and flexibility in managing network communications, helping ensure a smoother and more predictable experience for both clients and backends. For additional details and examples, refer to the official timeouts API documentation.
+
+## Gateway Infrastructure Labels
+
+While Gateway API providers a common API for different implementations, each implementation will have different resources created under-the-hood to apply users intent.
+This could be configuring cloud load balancers, creating in cluster Pods and Services, or more.
+
+While the API has always provided an extension point -- `parametersRef` in `GatewayClass` -- to customize implementation specific things, there was no common core way to express common infrastructure customizations.
+
+Gateway API v1.0 paves the way for this with a new `infrastructure` field on the `Gateway` object, allowing customization of the underlying infrastructure.
+For now, this starts small with two critical fields: labels and annotations.
+When these are set, any generated infrastructure will have the provided labels and annotations set on them.
+
+For example, I may want to group all my resources for one application together:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: hello-world
+spec:
+  infrastructure:
+    labels:
+      app.kubernetes.io/name: hello-world
+```
+
+In the future, we are looking into more common infrastructure configurations, such as resource sizing.
+
+## Support for Websockets, HTTP/2 and more!
+
+## `gwctl`, our new Gateway API command line tool
```

Comment:
````suggestion
## `gwctl`, our new Gateway API command line tool

`gwctl` is a command line tool that aims to be a `kubectl` replacement for viewing Gateway API resources.

The initial release of `gwctl` that comes bundled with Gateway v1.0 release includes helpful features for managing Gateway API Policies. Gateway API Policies serve as powerful extension mechanisms for modifying the behaviour of Gateway resources. One challenge with using policies is that it may be hard to discover which policies are affecting which Gateway resources. `gwctl` helps bridge this gap by answering questions like:

* Which policies are available for use in the Kubernetes cluster?
* Which policies are attached to a particular Gateway, HTTPRoute, etc?
* If policies are applied to multiple resources in the Gateway resource hierarchy, what is the effective policy that is affecting a particular resource? (For example, if an HTTP request timeout policy is applied to both an HTTPRoute and its parent Gateway, what is the effective timeout for the HTTPRoute?)

`gwctl` is still in the very early phases of development and hence may be a bit rough around the edges.

### Try it out!

```bash
# Clone the Gateway API repository.
git clone https://github.com/kubernetes-sigs/gateway-api.git

# Go to the gwctl directory.
cd gateway-api/gwctl

# Ensure vendor depedencies.
go mod tidy
go mod vendor

# Build the gwctl binary.
go build -o bin/gwctl cmd/main.go

# Add binary to PATH.
export PATH=./bin:${PATH}

# Start using!
gwctl help
```

### Examples
Here are some examples of how `gwctl` can be used:

```bash
# List all policies in the cluster. This will also give the resource they bind
# to.
gwctl get policies -A

# List all available policy types.
gwctl get policycrds

# Describe all HTTPRoutes in namespace ns2. (Output includes effective policies)
gwctl describe httproutes -n ns2

# Describe a single HTTPRoute in the default namespace. (Output includes
# effective policies)
gwctl describe httproutes my-httproute-1

# Describe all Gateways across all namespaces. (Output includes effective
# policies)
gwctl describe gateways -A

# Describe a single GatewayClass. (Output includes effective policies)
gwctl describe gatewayclasses foo-com-external-gateway-class
```

````